doc/channel-plugin.md

```diff
@@ -0,0 +1,59 @@
+## Channel Plugin
+
+Channel plugins are processes that runs continuously and independently of each other. They receive many requests over
+their lifetime.
+They receive JSON-formatted requests on stdin and reply with JSON-formatted responses on stdout. The request and
+response structure is inspired by JSON-RPC.
```

Comment:
The line break in the middle of the paragraph looks somewhat strange, I'd just continue the line (at least I fail to see a pattern why you'd add a line break there)